state/charm.go

```diff
@@ -737,7 +746,15 @@ func validateCharmSeries(modelType ModelType, series string, ch hasMeta) error {
 	if series == "" {
 		return nil
 	}
-	return model.ValidateSeries(model.ModelType(modelType), series, ch.Meta().Format())
+
+	// TODO (manadart 2021-04-15): This is copied from ReadCharm in the charm
+	// package. It should be encapsulated there instead of repeated.
+	format := charm.FormatV2
+	if ch.Manifest() == nil || len(ch.Manifest().Bases) == 0 {
+		format = charm.FormatV1
+	}
+
+	return model.ValidateSeries(model.ModelType(modelType), series, format)
```

Comment:
This is actually wrong... we should re-write this and `model.ValidateSeries` in a follow up commit.

Reply:
https://github.com/juju/juju/pull/12888